crates/rpc-client/src/rpc_client.rs

```diff
@@ -728,6 +730,277 @@ impl RPCClient {
+            filter: None,
+        };
+        let order = Order::Desc;
+        let limit = Uint32::from(DEFAULT_QUERY_LIMIT as u32);
+
+        let mut ckb_candidates = CandidateCustodians::default();
+        let mut sudt_candidates = HashMap::new();
+        let mut candidate_cells = 0usize;
+        let mut candidate_capacity = 0u128;
+        let mut candidate_fulfilled_sudts = 0usize;
+        let mut cursor = None;
+
+        // withdrawal ckb + change custodian capacity
+        let required_capacity = {
+            let withdrawal_capacity = withdrawals_amount.capacity;
+            withdrawal_capacity.saturating_add(custodian_change_capacity)
+        };
+
+        while candidate_capacity < required_capacity
+            || candidate_fulfilled_sudts < withdrawals_amount.sudt.len()
+            || candidate_cells < max_custodian_cells
+        {
+            let cells: Pagination<Cell> = to_result(
+                self.indexer
+                    .client()
+                    .request(
+                        "get_cells",
+                        Some(ClientParams::Array(vec![
+                            json!(search_key),
+                            json!(order),
+                            json!(limit),
+                            json!(cursor),
+                        ])),
+                    )
+                    .await?,
+            )?;
+
+            if cells.last_cursor.is_empty() {
+                break;
+            }
+            cursor = Some(cells.last_cursor);
+
+            for cell in cells.objects.into_iter() {
+                let args = cell.output.lock.args.clone().into_bytes();
+                let custodian_lock_args = match CustodianLockArgsReader::verify(&args[32..], false)
+                {
+                    Ok(()) => CustodianLockArgs::new_unchecked(args.slice(32..)),
+                    Err(_) => continue,
+                };
+
+                if custodian_lock_args.deposit_block_number().unpack() > last_finalized_block_number
+                {
+                    continue;
+                }
+
+                let opt_sudt_type_script = cell.output.type_.clone().map(|json_script| {
+                    let script = ckb_types::packed::Script::from(json_script);
+                    Script::new_unchecked(script.as_bytes())
+                });
+
+                // Verify sudt
+                if let Some(sudt_type_script) = opt_sudt_type_script.as_ref() {
+                    // Invalid custodian type script
+                    let l1_sudt_script_type_hash =
+                        rollup_context.rollup_config.l1_sudt_script_type_hash();
+                    if sudt_type_script.code_hash() != l1_sudt_script_type_hash
+                        || sudt_type_script.hash_type() != ScriptHashType::Type.into()
+                    {
+                        continue;
+                    }
+
+                    if sudt_type_script.hash() != CKB_SUDT_SCRIPT_ARGS
```

Comment:
Please explain this line, why sudt_type_script should equals to CKB_SUDT_SCRIPT_ARGS?

Reply:
Fixed. Now use enum to represent custodian type.